packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -2791,6 +2791,12 @@ class _RectangularSliderValueIndicatorPathPainter {
     double scale,
   }) {
     assert(!sizeWithOverflow.isEmpty);
+    /// If parentBox is null this means that the Slider was removed without
+    /// disposing value indicator.
```

Comment:
We're not checking to see if parentBox is null here.  More importantly, I assume we're failing to do some work in _RenderSlider.detach: when the Slider is detached, we should be removing the value indicator's overlay entry. In other words, we shouldn't end up here if the slider is detached.

Reply:
I tried removing the value indicator from the slider detach method but that wouldn't work because it would still appear as attached by the time it got to this part of the code. 

It was similar to the lag issue we were having with the overlay and so was the solution. Instead of checking if the parentbox is attached I am checking from the paint methods with an overlay. 

This is easier to understand through the commit changes.

https://github.com/flutter/flutter/pull/57535/commits/6b4636afcdc7f8851786e3dabd9d8e141dc9c885